### PR TITLE
Merged DBV2 code from Master19 into daffy

### PR DESCRIPTION
**Pull request by AmeliaEScott**
This code was written by Merlin Hosner (@hosnerm), with minor changes by Timothy Scott (@ItsTimmy) when porting to daffy.

### CODE REVIEW

File: packages/duckiebot_interface_dbv2/launch/all_drivers.launch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <arg name="robot_type" default="duckiebot" doc="The type of the robot: should be duckiebot, watchtower, or traffic_light"/>
+
+    <!-- Camera Node: reads frames from a physical camera and produces ROS messages -->
+    <include file="$(find camera_driver)/launch/camera_node.launch">
+        <arg name="veh" value="$(arg veh)"/>
+        <arg name="param_file_name" value="$(arg robot_type)"/>
+    </include>
+
+    <!-- LED Emitter: reads ROS messages and commands the LEDs -->
+    <group unless="$(eval arg('robot_type') == 'watchtower')">
```
Comment (AmaurX) [inline packages/duckiebot_interface_dbv2/launch/all_drivers.launch]: I guess we wont have watchtowers v2 yet right? maybe get rid of all "watchtower" logic here ?

File: packages/duckiebot_interface_dbv2/launch/all_drivers.launch [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        <include file="$(find led_emitter)/launch/led_emitter_node.launch">
+            <arg name="veh" value="$(arg veh)"/>
+            <arg name="robot_type" value="$(arg robot_type)"/>
+        </include>
+    </group>
+
+
+    <!-- Joystick: reads from a physical joystick and produces ROS messages -->
+    <group if="$(eval arg('robot_type') == 'duckiebot')">
```
Comment (AmaurX) [inline packages/duckiebot_interface_dbv2/launch/all_drivers.launch:21]: Also, wont the robot type be something else than duckiebot ? If eventually we want everything together then it should have a new name

File: packages/duckiebot_interface_dbv2/launch/all_drivers.launch [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <!-- Wheel Drivers: reads ROS messages and spins the motors -->
+        <include file="$(find wheels_driver_dbv2)/launch/wheels_driver_node.launch">
+            <arg name="veh" value="$(arg veh)"/>
+            <arg name="param_file_name" value="$(arg param_file_name)"/>
+        </include>
+
+   </group>
+
+    <!-- Traffic Light (Traffic Light robot only): commands the LEDs of a traffic light -->
+    <group if="$(eval arg('robot_type') == 'traffic_light')">
```
Comment (AmaurX) [inline packages/duckiebot_interface_dbv2/launch/all_drivers.launch]: same comment as for the watchtowers

File: packages/duckiebot_interface_dbv2/package.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<package>
+  <name>duckiebot_interface_dbv2</name>
+  <version>1.0.0</version>
+  <description>The duckiebot_interface meta package</description>
+
+  <license>GPLv3</license>
+<maintainer email="mack@duckietown.org">Mack</maintainer>
```
Comment (AmaurX) [inline packages/duckiebot_interface_dbv2/package.xml]: real email address ?

File: packages/wheels_driver_dbv2/package.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>duckietown_msgs</build_depend>
+  <build_depend>rospy</build_depend>
+  <build_depend>adafruit_drivers</build_depend>
+
+  <run_depend>duckietown_msgs</run_depend>
+  <run_depend>rospy</run_depend>
+  <run_depend>adafruit_drivers</run_depend>
```
Comment (AmaurX) [inline packages/wheels_driver_dbv2/package.xml]: for shorter files, I think just "< depend >BALBLA</ depend>" is enough (it covers both the build and the run depends, and those should be used only when only one is needed)

File: packages/wheels_driver_dbv2/setup.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+## ! DO NOT MANUALLY INVOKE THIS setup.py, USE CATKIN INSTEAD
```
Comment (AmaurX) [inline packages/wheels_driver_dbv2/setup.py:1]: then is this file needed ?

**Review by AmaurX — COMMENTED** [review]
Did not really read the code, but more the rest of the files for clarity and future integration

File: packages/wheels_driver_dbv2/setup.py (hunk 6 above)
Comment (AmeliaEScott) [reply packages/wheels_driver_dbv2/setup.py:1]: Yes, it is used by catkin

File: packages/duckiebot_interface_dbv2/launch/all_drivers.launch [hunk 7]
```diff
@@ -35,12 +33,10 @@
...
-    <group if="$(eval arg('robot_type') == 'traffic_light')">
-        <remap from="led_emitter_node/custom_pattern" to="~custom_pattern"/>
-        <include file="$(find traffic_light)/launch/traffic_light_node.launch">
-            <arg name="veh" value="$(arg veh)"/>
-        </include>
-    </group>
+    <remap from="led_emitter_node/custom_pattern" to="~custom_pattern"/>
+    <include file="$(find traffic_light)/launch/traffic_light_node.launch">
+        <arg name="veh" value="$(arg veh)"/>
+    </include>
```
Comment (AmaurX) [inline packages/duckiebot_interface_dbv2/launch/all_drivers.launch]: @ItsTimmy this whole part should be removed, as it is only for traffic lights !

**Review by AmaurX — CHANGES_REQUESTED** [review]
